src/helm/proxy/clients/critique_client.py

```diff
@@ -209,3 +212,155 @@ def make_critique_request(self, request: CritiqueRequest) -> CritiqueRequestResu
         task_id = self._get_or_create_task(project_id, request.fields)
         worker_responses = self._get_worker_responses(task_id, request.template.questions)
         return CritiqueRequestResult(worker_responses)
+
+
+class LLMCritiqueClient(CritiqueClient):
+    """A CritiqueClient that queries a LLM to answer CritiqueRequests."""
+
+    def __init__(self, client: Client, model_name: str, cache_config: CacheConfig):
+        self._client = client
+        self._model_name = model_name
+        self._cache = Cache(cache_config)
+
+    def _interpolate_fields(self, text: str, fields: Dict[str, str]) -> str:
+        for key, value in fields.items():
+            text = text.replace("{{" + key + "}}", value)
+        return text
+
+    def _question_to_prompt(self, question: CritiqueQuestionTemplate, fields: Dict[str, str]) -> str:
+        prompt: str = self._interpolate_fields(question.text, fields)
+        if question.question_type == "free_response":
+            prompt += "\nAnswer: "
+        else:
+            prompt += "\nAnswer options: "
+            for i, letter in enumerate(string.ascii_uppercase[: len(question.options)]):
+                prompt += f"\n{letter}. {question.options[i]}"
+            if question.question_type == "multiple_choice":
+                prompt += "\nAnswer with a single letter corresponding to the option.\nAnswer: "
+            elif question.question_type == "checkbox":
+                prompt += (
+                    "\nAnswer with all letters seperated by commas, corresponding to the selected options.\nAnswer: "
+                )
+        return prompt
+
+    def _task_to_requests(self, task: CritiqueTaskTemplate, fields: Dict[str, str]) -> List[Request]:
+        base_prompt: str = self._interpolate_fields(task.instructions, fields)
+
+        requests: List[Request] = []
+        for question in task.questions:
+            prompt: str = base_prompt + "\n\n" + self._question_to_prompt(question, fields)
+            max_tokens = 1
```

Comment:
Put this at the end as `else` so you don't have to override variables (make this code more functional)